File: HRMS_Modular_Monolithic_BolierPlate Without Git/Modules/AnnouncementFeature/AnnouncementFeature.Domain/Announcement.cs
using HRMS.Core.Postgres.Common;
using HRMS.Shared.Domain.Entity;

namespace AnnouncementFeature.Domain
{
    public class Announcement : BaseEntity
    {
        public string? Title { get; set; }
        public string? Category { get; set; }
        public string? Priority { get; set; }
        public string? Content { get; set; }
        public string? VisibilityScope { get; set; }
        public DateTime? ExpiryDate { get; set; }
        public int? Views { get; set; }
        public int? Acknowledgements { get; set; }
        public string? CreatedBy { get; set; }
        public DateTime? CreatedAt { get; set; }
        public string DocumentType { get; set; } = nameof(Announcement);
        public UserBase? UserContext { get; set; }
    }
}

File: graphql/query/announcement.ts
import { gql } from "@apollo/client";

export const GET_ALL_ANNOUNCEMENT = gql`
  query GetAllAnnouncement($request: GetAllAnnouncementRequestInput!) {
    getAllAnnouncement(request: $request) {
      data {
        announcement {
          id
          announcementId
          title
          category
          priority
          content
          visibilityScope
          expiryDate
          views
          acknowledgements
          createdBy
          createdAt
        }
      }
      meta {
        totalCount
        skip
        take
      }
      success
      message
      statusCode
    }
  }
`;
